server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -396,7 +396,13 @@ private TableMgmtStats manageTablets(Iterator<TabletManagement> iter,
       }
 
       if (tm.getOperationId() != null) {
-        goal = TabletGoalState.UNASSIGNED;
+        // If there are still wals the tablet needs to be hosted
+        // to process the wals before starting the op
+        if (tm.getLogs().isEmpty()) {
+          goal = TabletGoalState.UNASSIGNED;
+        } else {
```

Comment:
May want to limit this to hosting on merge operations with wals, so if opid is merge and it has wals then host.  Eventually split will need it too (#3844), but may be good to make the FATE aware of logs.  Delete operations probably will never need tablets hosted for recovery.